src/content/docs/en/developers/transaction-fees-on-scroll.mdx

```diff
@@ -165,10 +165,26 @@ Computes the amount of L1 gas used for a transaction. Adds the overhead which re
 | --------- | ----------------------------------------------------------- |
 | data      | Signed fully RLP-encoded transaction to get the L1 fee for. |
 
+## L1 Originated Transactions
+
+When messaging from L1 to L2, the user does not pay the L1 fee but may need to know how much L2 gas to pay. Contracts on L1 can use an L2 Gas Price Oracle deployed to the L1 to get the gas fee for a given transaction. The oracle also exposes the current l2BaseFee and estimated cross-domain message fee for a given gas limit.
```

Comment:
Nitpick but "does not pay the L1 fee" is a bit misleading. Better phrasing would be "the user pays all fees on L1". And then talk about "L1 gas fee" (not part of Scroll) + bridging fee (that pays for L2 execution)

Reply:
adjusted.